src/DefaultNodeTypes.jl

```diff
@@ -134,6 +186,32 @@ function (s::LinearConditional)(res::Array{Float64},
   nothing
 end
 
+# parametric specific functor
+function (s::LinearConditional{<:ParametricTypes})(X1::Vector{Float64},
```

Comment:
I'm not fond of the parametric specific functors (the code maintainability cost will be way too high), rather do a wrapper on the existing so that functors are defined only once according to the existing standard:
https://www.juliarobotics.org/Caesar.jl/latest/concepts/adding_variables_factors/
https://www.juliarobotics.org/Caesar.jl/latest/examples/basic_definingfactors/

Reply:
Noted. Reasons for it was in slack, can't remember it now. 
I would suggest we keep it as is for now since it works. We can add tests and then swap over to ensure it works the same.
I'm slightly worried about the performance of the mm functors for parametric.

Reply:
Right, so if there is a performance hit with the mm functions, that needs to be fixed immediately!  I spent a LOT of time on those functions to get good performance for where things were.  There are one or two memory things that can be better for the general `multihypo` interface, but that is a layer higher than what you see here, so should be fine.

i try get wrappers to you ASAP, because otherwise these things float far apart -- maybe little bit of slowly in the right direction type of argument.

Reply:
How would you feel about removing the `idx` parameter and only working with the vector (not the matrix). It makes it a bit easier and improves readability.

Reply:
#467